treasury-factory/src/lib.rs

```diff
@@ -56,17 +60,95 @@ impl Contract {
+        create_dao_args: String,
+    ) -> Promise {
+        if env::predecessor_account_id() != env::current_account_id() {
+            env::panic_str("Should not be called directly");
+        }
+        let create_account_result = env::promise_result(env::promise_results_count() - 1);
+        let create_account_result: bool = match create_account_result {
+            PromiseResult::Successful(result) => {
+                near_sdk::serde_json::from_slice::<bool>(&result).unwrap_or(false)
+            }
+            _ => false,
+        };
+
+        if create_account_result {
+            Promise::new(new_instance_contract_id.clone())
+                .then(
+                    sputnik_dao::ext(sputnik_dao_factory_account_id.parse().unwrap())
+                        .with_attached_deposit(CREATE_SPUTNIK_DAO_DEPOSIT)
+                        .with_static_gas(Gas::from_tgas(100))
+                        .create(name.to_string(), create_dao_args),
+                )
+                .then(Self::ext(env::current_account_id()).create_dao_callback(
+                    refund_on_failure_account,
+                    new_instance_contract_id,
+                    format!("{}.{}", name, sputnik_dao_factory_account_id),
+                    social_db_account_id,
+                    widget_reference_account_id,
+                ))
+        } else {
+            env::log_str(
+                format!(
+                    "Failed creating treasury web4 account {}",
+                    new_instance_contract_id
+                )
+                .as_str(),
+            );
+            Promise::new(refund_on_failure_account).transfer(refund_on_failure_amount)
+        }
+    }
+
+    pub fn create_dao_callback(
+        &self,
+        refund_on_failure_account: AccountId,
+        new_instance_contract_id: AccountId,
+        sputnik_dao_contract_id: String,
+        social_db_account_id: String,
+        widget_reference_account_id: String,
+    ) -> Promise {
+        if env::predecessor_account_id() != env::current_account_id() {
+            env::panic_str("Should not be called directly");
+        }
+        let create_dao_result = env::promise_result(env::promise_results_count() - 1);
+
+        match create_dao_result {
+            PromiseResult::Successful(_result) => Promise::new(new_instance_contract_id.clone())
+                .then(
+                    instance_contract::ext(new_instance_contract_id.clone())
+                        .with_attached_deposit(NearToken::from_near(1))
+                        .update_widgets(widget_reference_account_id, social_db_account_id),
+                ),
+            PromiseResult::Failed => {
+                env::log_str(format!("Succeeded creating and funding web4 account {}, but failed creating treasury account {}.",
+                    new_instance_contract_id, sputnik_dao_contract_id).as_str());
+                Promise::new(refund_on_failure_account).transfer(CREATE_SPUTNIK_DAO_DEPOSIT)
```

Comment:
I believe it should refund 7 NEAR since SocialDB deposit does not even get scheduled, right?